agent-filtering.md

```diff
@@ -0,0 +1,28 @@
+# Filter interface
+
+The filtering capability allows running arbitrary filter at the request entry point to the application.
+The instrumentation points use the result of the evaluation to block the execution.
+
+## Filter interface
+
+The filter interface defines following API for request evaluation. Each of these methods
+can be called multiple times.
+
+### `boolean evaluateRequestHeaders(Span, Map<String, String>)`
+
+It evaluates request headers.
+
+### `boolean evaluateRequestBody(Span, String body)`
+
+It evaluates request body.
+
```

Comment:
Can we add `boolean evaluateRequest(Span, Map<String, String> headers,  String body)`? Some evaluators might need both headers and body for evaluations?

Reply:
+1 added